### PR TITLE
fix: clean up managed default versions

### LINKED ISSUE #11561: package.json complains about "construct-style-sheets-polyfill" version
**Describe the bug**
The `package.json` file contains multiple dependency versions of `construct-style-sheets-polyfill` (differing between dev and non-dev) causing IntelliJ to display the following error message:
```
The installed version "construct-style-sheets-polyfill@2.4.6" doesn't match the version range "2.4.16"
```

The `construct-style-sheets-polyfill` dependencies found in the `package.json` file are (obviously abbreviated):
```
{
  "dependencies": {
    "construct-style-sheets-polyfill": "2.4.16"
  },
  "devDependencies": {
    "construct-style-sheets-polyfill": "2.4.6",
  },
  "vaadin": {
    "dependencies": {
      "construct-style-sheets-polyfill": "2.4.16",
    },
    "devDependencies": {
      "construct-style-sheets-polyfill": "2.4.6",
    },
  }
}
```

**To Reproduce**
1. Upgrade maven project dependency to use Vaadin Flow 14.7.0.alpha3
2. Right-click on project (or pom.xml) and select Maven > Reload Project
3. IntelliJ displays notification about updating npm.
4. Updating npm does not resolve versions, and IntelliJ redisplays notification about updating npm.
5. Removing the `construct-style-sheets-polyfill` dependencies and rebuilding restores them as shown above.

**Expected behavior**
Consistent versions, no warnings.

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.5.1
 - Browser: N/A
 - Version: N/A
 - Vaadin platform version: 14.7.0.alpha3

**Additional context**
Upgraded project from 14.6.1, but don't think that is relevant.

### PR DESCRIPTION
**Pull request by caalador**
Clean up default dependencies and
devDependencies if they have changed.

Fixes #11561

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java [hunk 1]
```diff
@@ -191,26 +192,28 @@ private boolean updatePackageJsonDependencies(JsonObject packageJson,
                         getDefaultDependencies().entrySet().stream())
                 .map(Entry::getKey).collect(Collectors.toList());
 
-        JsonObject vaadinDependencies = packageJson.getObject(VAADIN_DEP_KEY)
-                .getObject(DEPENDENCIES);
         boolean doCleanUp = forceCleanUp;
         int removed = removeLegacyProperties(packageJson);
+        removed += cleanDependencies(dependencyCollection, packageJson,
```
Comment (pleku) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:197]: I wonder if my change #10864 should be picked on top of this too 🤔

**Review by pleku — APPROVED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java (hunk 1 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:197]: Perhaps. But I haven't seen any issues on that front for the 14 branch as here we don't have the fusion dependencies etc. which complicate the dependency tree.

Comment (pleku) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:197]: I'm not sure how fusion comes to effect. I'm thinking about a simple case where
1) one is using 14.7.0 + npm
2) 14.7.1 is released
3) transitive dependencies jump for components, like vaadin-overlay jump to 14.7.1 automatically
-> duplicates appear as vaadin-dialog stays in 14.7.0 ❓ 

Or was there something that prevents this from happening in 14-series ?

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java:197]: Probably nothin apart from shrinkwrap.

**Pull request merged**